### PR TITLE
[Feat] Use rawTransaction

### PR DESCRIPTION
**Pull request by karlfloersch**
# Overview

Addresses https://github.com/ethereum-optimism/roadmap/issues/757

Make the batch submitter use `rawTransaction` instead of doing crazy encoding.

### CODE REVIEW

File: src/batch-submitter/tx-batch-submitter.ts
Comment (karlfloersch) [inline src/batch-submitter/tx-batch-submitter.ts:666]: probably rename rawTransaction

File: src/types.ts
Comment (karlfloersch) [inline src/types.ts:47]: Remove EIP155TxData type

**Comment by karlfloersch** [comment]
TODO: update package.json to use new provider once merged

File: src/batch-submitter/tx-batch-submitter.ts
Comment (annieke) [inline src/batch-submitter/tx-batch-submitter.ts:688]: TODO: update block type to contain rawTransaction

File: test/batch-submitter/mockchain-provider.ts
Comment (karlfloersch) [inline test/batch-submitter/mockchain-provider.ts:87]: todo!

File: test/batch-submitter/mockchain-provider.ts
Comment (karlfloersch) [inline test/batch-submitter/mockchain-provider.ts:110]: NIce!! We should totally be able to remove both of these now!

File: package.json
Comment (karlfloersch) [inline package.json:32]: Oh can we actually remove this dependency now that we're doing the raw `.send(..)` calls?

**Review by karlfloersch — COMMENTED** [review]
Awesome! There were a few comments left around but overall this PR is already looking quite good! Didn't look into why CI is failing but I'd imagine something super simple

File: src/utils.ts
Comment (karlfloersch) [inline src/utils.ts:25]: Should this be `num: string`?

Is there a function which does this in `core-utils`? If not I think it's fine to include here but eventually probably a good idea to move it there

**Review by karlfloersch — COMMENTED** [review]
Left one nit but this looks good! 

Oh I also can't approve this PR cuz I created it whoops

File: src/utils.ts
Comment (annieke) [reply src/utils.ts:25]: i copied this from ethers formatter, which i think is supposed to keep the input general since it may begin as a number too

https://github.com/ethers-io/ethers.js/blob/6c43e20e7a68f3f5a141c74527ec63d9fe8458be/packages/providers/src.ts/formatter.ts#L165-L170

i'll move it into core-utils in a followup PR :)

**Review by annieke — APPROVED** [review]
lgtm 🥇 💯

**Pull request merged**
